andi-bigswitch/pybsn
# bsc 14439 precommit py

- pybsn-schema: add support for RPC nodes
- pybsn-schema: add support to load json from file
- python 2/3 compatibility
- pybsn-repl: fix schema display
- pybsn/attempt_login: fix host discovery fallback
- pybsn-repl: Update for IPython 7
- README: add basic info on pybsn-repl
- pybsn-repl: correctly deal with empty containers
- README: no longer working reference to codeship
- README: fix typo
- PyBSN: add support for RPC calls
- Examples/support_bundle: use RPC
- Bump version and fix typos in distribution file
- Update README.md
- BCF/API; some remaining py3 compatibility fixes
- BSC-12610: fix broken unit tests
- .gitignore: ignore VIM swap files
- BSC-12611: pybsn - support API tokens
- add pre-commit hook
- pybsn-repl: add some debug logging
- connect: do not validate tls by default
- Bump version to 0.2.0
- Add initial change log
- Update README.md
- BSC-12819: for schema show of a RPC, recurse into full schema
- BSC-12819: pybsn-repl; show schema - correct max_depth recursion
- BSC-12875: handle return 204s from RPCs correctly
- BSC-12875: add unit tests for RPC and connect
- pybsn: log under "pybsn"; log request+response bodies
- pybsn-repl: allow more fine-grained verbosity
- Bump version to 0.3.0 for API change to connect
- test_switch: remove use of betamax, responses instead
- pybsn: Fix py2 bug with unicode strings in predicates
- pybsn: add guard for debug request logging
- test_switch: deal with sometimes-escaping of []
- pybsn: reenable py2 test build
- Update changelog for version 0.3.0
- enable using BigDB Client as a context manager, close session on exit
- pybsn: fix close when no token is available, add unit test
- show coverage for pybsn/__init__ (core pybsn)
- test_client: REs to deal with quoting issue
- MCD-373: allow specifying request/session headers on connect
- MCD-373: Bump version to 0.3.1; add change log
- Update Repo in README.md
- Update README.md
- Improve PyBSN docs
- address comments
- address comments
- Update README.md with mutation examples
- BSC-13395: use login RPC for modern controllers
- BSC-13395: adapt to login RPC using HTTP status code for error messaging
- Allow --env-token/-e to allow loading token from environment variable
- bump version to 3.2, update changelog
- improve formatting
- fix broken unittest + add case for modern RPC login
- fix test_close_no_auth test (was not doing anything)
- import urlparse dynamically to restore python2 compatibility
- install ipython and traitlets as package dependencies
- BSC-13695: remove partial BCF "API" porcelain layer
- BSC-13695: convert to UNIX lineendings
- BSC-13695: fix some quotes in examples
- BSC-14339: pybsn-repl; param to execute a non-interactive command with -c
- bump version to 0.3.3, remove no longer available author emails
- add release note for 0.3.3
- Update CHANGELOG.md
- Replace Travis CI with Github Action, test in os matrix
- Add action to release to github when tag is pushed
- Add action to release to PyPi when tag is pushed
- Properly select twine repo based on user input
- Release 3.3.3
- Use ISO date format in changelog
- BSC-14372: drop nosetest, check syntax, code style and coverage
- BSC-14372: stop testing python 2 compatibility
- BSC-14372: conform with introduced style checks / PEP8
- BSC-14372: ignore coverage reports, remove not needed config file
- BSC-14372: add CONTRIBUTORS.md, remove py2 mentions
- BSC-14372: remove source hacks for py2 compat. bump to 0.4.0
- BSC-14372: fix whitespace style violations
